client/core/core.go

```diff
@@ -1822,10 +1825,10 @@ fetchers:
 		if !wallet.connected() {
 			continue
 		}
-		if !c.cfg.NoAutoWalletLock {
+		if !c.cfg.NoAutoWalletLock && wallet.unlocked() { // no-op if Logout did it
```

Comment:
RE: @buck54321  https://github.com/decred/dcrdex/pull/2781#issuecomment-2146366759

We get that error log because we fail to check that the wallet is locked and attempt to lock it again. This affects both wallet lock in `Logout` and `Run`.

`Logout` and `Run` do not run in parallel, `Logout` would finish first before `Run` exits.

1. User presses ctrl+c. 
2. We catch that and `promptShutdown`. Here we attempt to `Logout`  which will try to lock wallets. 
3. if `Logout` is successful, the context used by `Run` is cancelled and before `Run` exits, it'll also try to lock wallets.

Both wallet lock sites do not check if the wallet needs to be locked, this PR fixes that.